### PR TITLE
 PP-10343 Add progress indicator for Task-list

### PR DESCRIPTION
**Pull request by AbdirizakIdris**
- Added function that calculates number of steps completed in the onboarding journey
- Added necessary tests for the function

### CODE REVIEW

File: app/controllers/your-psp/your-psp-tasks.service.js [hunk 1]
```diff
@@ -53,4 +53,13 @@ function stripeTaskListIsComplete (taskList) {
...
 }
 
-module.exports = { getStripeTaskList, stripeTaskListIsComplete }
+function stripeTaskListNumberOftasksComplete (taskList) {
+  const totalTasks = Object.keys(taskList).length
+  const completedTasks = Object.values(taskList).reduce(
+    (count, task) => count + task.completed,
+    0
+  )
+  return `${completedTasks} out of ${totalTasks}`
```
[the remark this answers is not in the document]
Comment (AbdirizakIdris) [reply app/controllers/your-psp/your-psp-tasks.service.js]: Yes that is a good Idea, updated now

File: app/controllers/your-psp/your-psp-tasks.service.js [hunk 2]
```diff
@@ -53,4 +53,15 @@ function stripeTaskListIsComplete (taskList) {
...
 
-module.exports = { getStripeTaskList, stripeTaskListIsComplete }
+function stripeTaskListNumberOftasksComplete (taskList) {
+  const totalTasks = Object.keys(taskList).length
+  const completedTasks = Object.values(taskList).reduce(
+    (count, task) => count + task.completed,
+    0
+  )
+  return {
+    no_of_tasks_completed: `${completedTasks}`,
```
Comment (kbottla) [inline app/controllers/your-psp/your-psp-tasks.service.js:63]:
```suggestion
    no_of_tasks_completed: completedTasks,
```

File: app/controllers/your-psp/your-psp-tasks.service.test.js [hunk 3]
```diff
@@ -204,5 +204,174 @@ describe('Stripe task list', () => {
       const taskList = getStripeTaskList(activeCredential, account)
       expect(stripeTaskListIsComplete(taskList)).to.equal(false)
     })
+
+    it('should correctly calculate when zero tasks are complete ', () => {
```
Comment (kbottla) [inline app/controllers/your-psp/your-psp-tasks.service.test.js:208]: Would be good to nest these tests into a separate describe as these are for a specific functionality `stripeTaskListNumberOftasksComplete`.

File: app/controllers/your-psp/your-psp-tasks.service.test.js [hunk 4]
```diff
@@ -204,5 +204,174 @@ describe('Stripe task list', () => {
...
+        organisationDetails: false,
+        governmentEntityDocument: false
+      }
+      const activeCredential = getCurrentCredential(account)
+      const taskList = getStripeTaskList(activeCredential, account)
+      expect(stripeTaskListNumberOftasksComplete(taskList).no_of_tasks_completed).to.equal('2')
+      expect(stripeTaskListNumberOftasksComplete(taskList).total_number_of_tasks).to.equal('7')
+    })
+
+    it('should correctly calculate when 3 tasks are complete ', () => {
```
Comment (kbottla) [inline app/controllers/your-psp/your-psp-tasks.service.test.js:273]: Don't think we need individual tests for all tasks. We can have 3 test cases covering

- all tasks completed
- none of the tasks completed
- few tasks completed

**Review by kbottla — APPROVED** [review]

**Pull request merged**